TrackMania 2D/src/tm2DGame/IPlayer.java

```diff
@@ -0,0 +1,71 @@
+package tm2DGame;
+
+import java.awt.Image;
+import java.awt.Rectangle;
+import java.awt.geom.Path2D;
+
+import ia.ga.impl.car.KeyEventGame;
+
+public interface IPlayer{
```

Comment:
Je reste un peu partagé pour cette interface. J'ai plutôt l'impression qu'il faudrait une classe abstraite (du coup CarComponent) car l'implémentation de certaines(presque toutes) de ces fonctions vont être la même pour n'importe quel player. En fait j'ai du mal à voir une autre impléméentation de IPlayer qui ne serait pas un carComponent ? Si c'est pour l'IA il va aussi reprendre la plupart de ses fonctions de CarComponent non ?

Reply:
ouai c'est pas faux

Reply:
en vrai faudrait un truc comme ça:

interface ICarPlayer::getAction() => renvoi un KeyEventGame

IaCarPlayer  implements ICarPlayer
     CarComponent car
     getAction() // implementation for IA

RealCarPlayer  implements IPlayer
     CarComponent car
     getAction() // implementation for a player